File: matlab/appendix_2_1_problem2_depth_152_5_to_180.m
function [d,I,peak,peak_d]=appendix_2_1_problem2_depth_152_5_to_180
% 附录2.1：第二问，d = 152.5:1:180 的分段积分计算。
[d,I]=x2;
[peak,i]=max(I);
peak_d=d(i);

File: python/appendix_2_1_problem2_depth_152_5_to_180.py
from __future__ import annotations

from x2 import x2


def run():
    """Appendix 2.1: problem 2, d = 152.5:1:180."""
    return x2()


if __name__ == "__main__":
    d, I, peak, peak_d = run()
    print("d =", d)
    print("I =", I)
    print("peak =", peak)
    print("peak_d =", peak_d)
